### ISSUE TITLE
CRDB: unable to run TPC-H

### ISSUE THREAD

**Issue by timveil** [timveil-cockroach/oltpbench#12]
```
15:29:42,503 (Worker.java:417) WARN  - PSQLException thrown when executing 'Q9/09' on 'TPCHWorker<000>' [Message='An I/O error occurred while sending to the backend.', ErrorCode='0', SQLState='08006']
org.postgresql.util.PSQLException: An I/O error occurred while sending to the backend.
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:335)
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:441)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:365)
	at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:143)
	at org.postgresql.jdbc.PgPreparedStatement.executeQuery(PgPreparedStatement.java:106)
	at com.oltpbenchmark.benchmarks.tpch.procedures.GenericQuery.run(GenericQuery.java:54)
	at com.oltpbenchmark.benchmarks.tpch.TPCHWorker.executeWork(TPCHWorker.java:43)
	at com.oltpbenchmark.api.Worker.doWork(Worker.java:386)
	at com.oltpbenchmark.api.Worker.run(Worker.java:294)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException
	at org.postgresql.core.PGStream.receiveChar(PGStream.java:308)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:1952)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308)
	... 9 more
15:29:42,504 (Worker.java:511) DEBUG - TPCHWorker<000> Q9/09 Result: UNKNOWN
15:29:42,504 (ThreadBench.java:769) ERROR - Unexpected fatal, error in 'TPCHWorker<000>' when executing 'Q9/09' [COCKROACHDB]
java.lang.RuntimeException: Unexpected fatal, error in 'TPCHWorker<000>' when executing 'Q9/09' [COCKROACHDB]
	at com.oltpbenchmark.api.Worker.doWork(Worker.java:545)
	at com.oltpbenchmark.api.Worker.run(Worker.java:294)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.postgresql.util.PSQLException: This connection has been closed.
	at org.postgresql.jdbc.PgConnection.checkClosed(PgConnection.java:783)
	at org.postgresql.jdbc.PgConnection.rollback(PgConnection.java:791)
	at com.oltpbenchmark.api.Worker.doWork(Worker.java:428)
	... 2 more

```

**Comment by timveil** [comment]
this is the problematic query...
```
select nation, o_year, sum(amount) as sum_profit from ( select n_name as nation, extract(year from o_orderdate) as o_year, l_extendedprice * (1 - l_discount) - ps_supplycost * l_quantity as amount from oltpbench.part, oltpbench.supplier, oltpbench.lineitem, oltpbench.partsupp, oltpbench.orders, oltpbench.nation where s_suppkey = l_suppkey and ps_suppkey = l_suppkey and ps_partkey = l_partkey and p_partkey = l_partkey and o_orderkey = l_orderkey and s_nationkey = n_nationkey and p_name like '%olive%' ) as profit group by nation, o_year order by nation, o_year desc
```

**Comment by timveil** [comment]
the closed/broken connection is actually caused by a `1m` timeout configured in haproxy.  that again means the underlying issue is query duration exacerbated by overly aggressive timeout in haproxy
```
Caused by: org.postgresql.util.PSQLException: This connection has been closed.
	at org.postgresql.jdbc.PgConnection.checkClosed(PgConnection.java:783)
	at org.postgresql.jdbc.PgConnection.rollback(PgConnection.java:791)
	at com.oltpbenchmark.api.Worker.doWork(Worker.java:428)
	... 2 more
```

**Comment by timveil** [comment]
resolved in `v2.2.0-alpha`

Also made the following configuration changes:
* increased `haproxy` timeout settings to `1h` to be save
* set `-max-sql-memory=4GB` on each node
